Lib/_android_support.py

```diff
@@ -77,14 +81,17 @@ def write(self, b):
             raise TypeError(
                 f"write() argument must be bytes-like, not {type(b).__name__}"
             ) from None
-        else:
-            b_out = bytes(b)
 
-        # Encode null bytes using "modified UTF-8" to avoid truncating the
-        # message.
-        b_out = b_out.replace(b"\x00", b"\xc0\x80")
+        b_out = bytes(b)
+        b_len = len(b_out)  # May be different from len(b) if b is an array.
```

Comment:
Thanks, that's a good simplification for the `bytes` case. There's no need to do a `type` check for the `str` case, because `str.__str__(s)` won't make a copy if `s `is already a `str`.

> I think that programmers that use Python on Android should be qualified enough to understand the original TypeError messages

In the BeeWare project we develop tools to make it as easy as develop on mobile as it is on desktop, and many of our users are beginners.